spec/factories/events.rb

```diff
@@ -0,0 +1,9 @@
+FactoryBot.define do
+    factory :event do
+        sequence(:name) { |n| "name#{n}" }
+        sequence(:description) { |n| "description#{n}" }
+        sequence(:discipline) { |n| "discipline#{n}" }
+        player_type Event.player_types[Event.player_types.keys.sample]
+        max_teams { rand(1..30) }
```

Comment:
Having a random number for this WILL lead to problems further down in testing abd make tests way more complex than they have to be. I'd put a static number of maybe 5 or 10.
Same with playertype.

Reply:
I just realized that we have random numbers in the matches factory aswell, out of my own experience with random data in tests I'm really not a fan of this. This can result in tests sometimes passing and sometimes failing depending on the random numbers and might be difficult to track down. So I'd still suggest removing these but I'll let you decide.

Reply:
I would leave them, as our tests should ideally be able to cope with different numbers.

Reply:
The problem with that is, that you have to essentially replicate the functionality that you are currently testing in your tests, instead of just hardcoding values. Which will make tests way more difficult to read and might also end up with you just copying the method you are currently testing. Which in the end defeats the purpose of testing at all.

Reply:
Since I have been requested for feedback, I will add some general thought without knowing the context of your solution and the specific test cases. 
Random tests are indeed a bit different than usual f(input)=output tests. However, if you apply them with caution, they can be the most valuable tests you could write. Since they do not test for specific values but some out of a large range, they usually have a larger input coverage and, therefore, can find more defects, which is the ultimate goal of testing. However, the input range should, in the ideal case, equal all possible user-defined inputs for the subroutine. Than, each failing test is a defect and SHOULD be resolved. Hence, I do nit agree with @Askir that sometimes passing and sometimes failing tests are bad. They are better than a test that is green since they identify misbehavior for some input. Sure, you have to output the chosen value to identify the bug. Otherwise the random value is useless. Random tests are especially useful if used as metamorphic tests (see testing slides or do individual research), when they define a relation between inputs and outputs instead of replicating the tested function. I did not look at your tests since no pointer was give to me. Hence I cannot say anything to your concrete implementation.
However, random tests with a small range of inputs (e.g., 1..30) might be transformed to a test that covers ALL of these inputs in one test, or in separate tests, depending on the kinds of inputs.